Governance updates

FredKSchott opened the pull request:
## Changes

This PR serves to implement several governance changes that will help our project as it continues to grow into its second year. The biggest changes (and the rationale behind them) are highlighted below:

1. **Remove technical requirement from Core.** This will let us recognize the contributions of more amazing leaders in our community who are not necessarily focused on technical contributions.
2. **Form a new Technical Steering Committee (TSC).** This group will take on the responsibility of technical decision making, RFC approvals, etc. that Core currently performs. Initially, current core members will become the founding TSC members.
3. **Simplify TSC consensus model.** Replace our current multi-stage consensus resolution flow with a simpler "rough consensus" model (made popular by the IETF). This is similar in practice to what we use today, but with more history, documentation and precedent.
4. **Move staff privileges to a new, more limited core membership.** This lets us empower anyone brought in to work on Astro full- or part-time, without letting them "skip the line" of our normal contributor ladder.
5. **Document additional project roles.** This lets us recognize teams like `@team-docs` and `@support-squad` more officially, and provide even more resources to them from a project governance perspective.
6. **Document the alumni role & process.** This will let us guide some absent maintainers into a new `@alumni` role (voluntarily), making room for new maintainers and core members.

aFuzzyBear commented on GOVERNANCE.md:
```diff
@@ -98,61 +102,45 @@ There is no strict minimum number of contributions needed to reach this level, a
+### Level 3 (L3) - Core
 
-A Core Maintainer has significant sway in software design decisions. For this reason, coding experience is critical for this role. Core Maintainer is the only level of contributor that does require a significant contribution history on GitHub.
+The **Core** role is available to community members who have a larger-than-usual impact on the Astro project and community. They are seen as leaders in the project and are listened to by the wider Astro community, often before they have even reached this level. A Core member is recognized for contributing a significant amount of time and energy to the project through issues, pull requests, bug fi ...
```
```suggestion
The **Core** role is available to community members who have a larger-than-usual impact on the Astro project and community. They are seen as leaders in the project and are listened to by the wider Astro community, often before they have even reached this level. A Core member is recognized for contributing a significant amount of time and energy to the project through issues, pull requests, bug fixes, implementing advanced enhancements/features, and/or activly posting on Discord. 
```

aFuzzyBear replied:
Just a wee nit,

FredKSchott replied:
thanks!

aFuzzyBear commented on GOVERNANCE.md:
```diff
@@ -76,12 +80,12 @@ There is no strict minimum number of contributions needed to reach this level, a
 - Ability to moderate Discord to remove spam, harmful speech, etc.
 - Ability to join the `@mods` role on Discord (optional, opt-in).
-- Ability to push branches to the repo (No more personal fork needed).
+- Ability to push branches directly to the `astro` GitHub repo (No more personal fork needed).
```
I HAve added those changes in, `astro` isnt a valid repo and this better reflects the Organisation on Github

FredKSchott replied:
yup, good call out

sarah11918 commented on GOVERNANCE.md:
```diff
@@ -1,31 +1,70 @@
+
+**Anyone** can become an Astro Contributor (yes, even you!). Regardless of skill, experience or background.
+
+Our goal is too always recognize all of our contributors and their effort given to Astro. 
```
```suggestion
Our goal is to always recognize all of our contributors and their effort given to Astro. 
```

aFuzzyBear replied:
🤦‍♂️ D'Oh

sarah11918 commented on GOVERNANCE.md:
```diff
@@ -1,31 +1,70 @@
+  Strive to go above and beyond our [Code of Conduct](CODE_OF_CONDUCT.md) and commit to a healthy, honest discourse when communicating: 
+  - Within our Github Project and its wider umbrella groups and projects.
+    - Pull Requests, 
+    - Issue's 
```
```suggestion
    - Issues 
```

sarah11918 commented on GOVERNANCE.md:
```diff
@@ -1,31 +1,70 @@
+All Contributor roles are granted for as long as the individual wishes to engage with the project. 
 
-All Contributor roles are granted for as long as the person wishes to engage with the project. A Contributor can voluntarily leave the project at any time (see [Retiring a Role](#retiring-a-role-alumni) below). In extreme cases -- such as a Code of Conduct violation -- this role may be revoked by a project Steward at their discretion.
+Contributor's can voluntarily leave the project at any time (see [Retiring a Role](#retiring-a-role-alumni) below). 
```
```suggestion
Contributors can voluntarily leave the project at any time (see [Retiring a Role](#retiring-a-role-alumni) below). 
```

sarah11918 commented on GOVERNANCE.md:
```diff
@@ -102,17 +149,21 @@ There is no strict minimum number of contributions needed to reach this level, a
+A Core member is recognized for contributing a significant amount of their time and energy to the project through issues, pull requests, bug fixes, implementing advanced enhancements/features, and is activly posting on Discord. 
 
-Not every contributor will reach this level, and that's okay! L2 Maintainers still have significant responsibility and privileges in our community.
+Not every contributor will reach this level, and that's okay! L2 Maintainers still have significant responsibility and privileges as is within our community.
```
```suggestion
Not every contributor will reach this level, and that's okay! L2 Maintainers still have significant responsibility and privileges within our community.
```

aFuzzyBear replied:
As in where did this come from.... like I have no excuses for this utter brain fart💨

FredKSchott commented:
Thanks for the reviews so far, everyone!

@aFuzzyBear I reverted some of the style changes that you pushed to the branch that I didn't agree with. As a general rule, you should always ask before pushing significant changes to someone else's open PR (obvious non-trivial fixes like a typo fix can be okay). This is especially true for governance changes where wording must be very intentional, but it is also a good rule for docs and code as well. When in doubt, ping first!

I'm happy to chat through anything that I reverted, and feel free to leave comments on anything that you'd still like to see changed. I also tried to keep unrelated changes to a minimum, so some reverted things I just considered out of scope for this PR.

jasikpark commented on GOVERNANCE.md:
```diff
@@ -175,39 +195,108 @@ The project Steward is currently: **@FredKSchott**
+Any Maintainer (L2 and above) can self-nominate by messaging the project Steward (`@steward`) on Discord.
 
-#### Responsibilities
+### Technical Steering Committee (TSC)
```
So does "member of the technical steering committee" end up being like a "super core member" in a sense, requiring expertise in the code base?

I'm sorry, I hadn't seen the reason for this change

matthewp replied:
The reason is so that people who are heavily involved but not working on the codebase can still become part of core. So the TSC becomes what the current core sort of was; for people working on the Astro codebase and who can guide the technical direction of the project.

FredKSchott replied:
yup, less "super core member" and more "one team of many". It is the only team right now with extra voting permissions in our goverance docs, but this aligns `tsc` closer to a project like `docs` or `support-squad`. 

It also opens the door for other teams to adopt their own special governance down the road, similar to TSC. For example, if there ever was a reason to add a voting mechanism for the docs team, we could define the docs voting system in this same place.

jasikpark replied:
hmm ok so trying to go in the direction of working groups, kinda?

tsc = team focused on the codebase
docs = team focused on the docs
support-squad = team focused on providing support in discord & github & more

jasikpark commented on GOVERNANCE.md:
```diff
@@ -175,39 +195,108 @@ The project Steward is currently: **@FredKSchott**
+
+Contributor roles are granted for as long as the person wishes to engage with the project. However, over time an active community member may choose to step away from the Astro project to work on other things. Moving on from a project is a natural and well-understood part of any open source community, and we celebrate it!
+
+**Alumni** is a special designation and role for any person who was once an active maintainer (L2 or above) but is now no longer actively involved. By retiring and joining Alumni you trade-in your current set of roles, privileges, and responsibilities for a new, special Alumni role (which comes with its own set of Privileges, as described above).
```
What is the role / privilege of being an Alumni? I imagine it's a way to communicate that you have participated in helping before, but that you're taking a break for some amount of time and aren't involved enough?

FredKSchott replied:
That's correct! No additional privileges other than what is outlined here.

jasikpark commented on GOVERNANCE.md:
```diff
@@ -318,17 +407,17 @@ If the pull request submitter is not a Core Maintainer, the PR can be closed by
+6. **If consensus is not reached:** The RFC author and TSC members must make all reasonable attempts to resolve issues and reach consensus in GitHub or a follow-up RFC meeting. The process of reaching consensus can take time, and should not be rushed as long as all participants are making a reasonable effort to respond.
+7. **If consensus still cannot be reached:** The project Steward may invoke [rough consensus](https://en.wikipedia.org/wiki/Rough_consensus) to resolve an RFC that has not achieved absolute consensus, as described below (borrowed from the [IETF](https://datatracker.ietf.org/doc/html/rfc2418)):
+
+> Working groups make decisions through a "rough consensus" process. Astro consensus does not require that all participants agree although this is, of course, preferred. In general, the dominant view of the TSC shall prevail. (However, "dominance" is not to be determined on the basis of volume or persistence, but rather a more general sense of agreement). Consensus can be determined by a show of  ...
```
Who determines rough consensus if an RFC is discussed on an RFC meeting w/o the Steward present?

I suppose that just comes down to what sort of RFC it is, how fundamental it is?

FredKSchott replied:
It's up the the steward to determine if rough consensus is reached, so they would have to be present in any case of non-absolute consensus.

In practice, all of our votes so far have been through absolute consensus (100%), and any objections have been addressable through follow-up and working together between the RFC champion and the objector. So this is a bit of an edge case, but still a good one to plan for.

jasikpark replied:
This extends to the "let's wait a moment for someone to object to consensus" done in RFC calls traditionally, right?

FredKSchott replied:
yup!

FredKSchott commented on GOVERNANCE.md:
```diff
@@ -175,39 +195,108 @@ The project Steward is currently: **@FredKSchott**
+- Invitation to the `staff` team on GitHub.
 
-When someone leaves the Astro Technology Company, they lose staff privileges and return to their original membership level in our governance structure (whatever level they were at before joining staff).
+Staff membership does not grant any additional abilities when it comes to voting and project governance. A Staff member is still eligible for other roles in the community and may still vote as defined by their other roles. For example, a Staff member who is also a part of `@core` will be able to vote as any other `@core` member would.
```
Not really! This flow you're describing is mainly covered by the "Core In Residence" role. 
For 99% of people, I imagine it will work like this:

- Someone joins The Astro Technology Company to work on the core codebase full-time
- They are nominated for "Core In Residence", meaning that they get the `@core` status but can not participate in voting.
- They leave the company and lose their core status. If they want to stick around our community, they can go through the normal nomination channels.

If someone joins the company but does not work in the community, they may get staff but wouldn't even be nominationed for maintainer, core, etc.

If someone joins the company and contributes to the codebase for many years, becoming a leader in the community, then they could have their "in residence" limitations removed by a normal core nomination vote.

Hope that helps clarify!

jasikpark replied:
Ah, so basically this is kinda switching the Staff role to a relatively decorative role + "core in residence" is an option to fast-track permissions for someone in the company but who hasn't necessarily been around long enough to earn any of the other statuses?

FredKSchott replied:
yup, exactly.

natemoo-re commented on GOVERNANCE.md:
```diff
@@ -43,30 +57,32 @@ Have you done something (big or small) to contribute to the health, success, or
 #### Nomination Process
 
-If you contributed to Astro outside of Discord, you may self-nominate by sending `!contribute` in any Discord channel, accompanied by a link or description of your contribution. You may also gain this role if you are active and helpful on Discord.
+You may self-nominate by sending the message `!contribute` in any Discord channel. If you do this, please share a second message with a link or description of your contribution so that people can recognize you for the contribution,
```
```suggestion
You may self-nominate by sending the message `!contribute` in any Discord channel. If you do this, please share a second message with a link or description of your contribution so that people can recognize you for the contribution.
```

natemoo-re reviewed (approved):
Big fan of these changes, approved! Saw one small typo.

jasikpark commented:
> Initially, current core members will become the founding TSC members.

Is this written down in the governance docs, or just we're going implicit w/ it?

FredKSchott commented:
>> Initially, current core members will become the founding TSC members.
>
> Is this written down in the governance docs, or just we're going implicit w/ it?

I don't think it needs to be, since we'd still be going through the normal nomination process to make this happen. But I'm happy to add a line about this to the governance doc to formalize it, if it feels like it would be helpful.

Yan-Thomas commented on GOVERNANCE.md:
```diff
@@ -175,39 +195,108 @@ The project Steward is currently: **@FredKSchott**
+- `@support-squad` runs the `#support-threads` channel and helps anyone who needs help using Astro.
 
-The staff role was designed to offer instant **visibility** and **trust**, but not instant **authority.**
+Many of these team roles can be be browsed and joined automatically by visiting the `#manage-roles` channel in our Discord. Getting involved with a team is a great way to start contributing to Astro!
```
A small typo I found:
```suggestion
Many of these team roles can be browsed and joined automatically by visiting the `#manage-roles` channel in our Discord. Getting involved with a team is a great way to start contributing to Astro!
```

FredKSchott replied:
thank you @Yan-Thomas!

jasikpark reviewed (approved):
Looks like a great plan to me! I'm excited to see how people can see their contributions reflected in their roles and responsibilities more with how this will be setup :)
